### PR TITLE
Dockerfile

### PR DESCRIPTION
**Pull request by Anakin100100**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM ruby:alpine
```
Comment (micorix) [inline Dockerfile:1]: Please pin image version

Comment (wojtodzio) [inline Dockerfile:1]: It would be nice to pin this image to a specific ruby version
```suggestion
FROM ruby:3.0.2-alpine
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+WORKDIR /code
+COPY . /code
```
Comment (micorix) [inline Dockerfile:3]: It could be a good thing to add dependencies first, install them and then add the rest to leverage docker layers caching. But we can deal with that later, when setting up CI deployment.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+RUN apk update && apk add --virtual build-dependencies build-base ruby-dev libpq postgresql-dev tzdata
```
Comment (micorix) [inline Dockerfile:4]: Please run apk update and apk add  at the begging of the Dockerfile. That way it'll get cached.

**Review by micorix — CHANGES_REQUESTED** [review]

**Comment by micorix** [comment]
Also please use imperative mood when wording PR names and commit messages

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+RUN bundle config set --local without 'development test'
+RUN bundle install 
+CMD rails s -b 0.0.0.0 -p 80 -e production
+
```
Comment (wojtodzio) [inline Dockerfile:8]: Dependencies are way less likely to change than the source code, and so it may be a good idea to put them close to the top so that docker can cache layers efficiently
```suggestion
WORKDIR /code
RUN apk update && apk add --virtual build-dependencies build-base ruby-dev libpq postgresql-dev tzdata
RUN bundle config set --local without 'development test'
COPY Gemfile Gemfile.lock ./
RUN bundle install 
COPY . /code
CMD rails s -b 0.0.0.0 -p 80 -e production
```

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FROM ruby:alpine
+WORKDIR /code
+COPY . /code
+RUN apk update && apk add --virtual build-dependencies build-base ruby-dev libpq postgresql-dev tzdata
+RUN bundle config set --local without 'development test'
+RUN bundle install 
+CMD rails s -b 0.0.0.0 -p 80 -e production
```
Comment (wojtodzio) [inline Dockerfile:7]: It would be nice to be able to easily override the default port

**Review by wojtodzio — CHANGES_REQUESTED** [review]
A few nitpicks :)

File: Dockerfile (hunk 5 above)
Comment (Anakin100100) [reply Dockerfile:9]: @wojtodzio When can build time port mapping input be needed?

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,10 @@
+FROM ruby:3.0.2-alpine
+RUN apk update
+RUN apk add --virtual build-dependencies build-base ruby-dev libpq postgresql-dev tzdata
+WORKDIR /code
+COPY Gemfile /code
+RUN bundle config set --local without 'development test'
```
Comment (wojtodzio) [inline Dockerfile:6]: I'm really nitpicking now, but you can put `WORKDIR` and `bundle config` to the very top, as those layers will likely never change

File: db/schema.rb [hunk 7]
```diff
@@ -12,57 +10,57 @@
...
+    t.text "concurrency_key"
+    t.text "cron_key"
+    t.uuid "retried_good_job_id"
+    t.datetime "cron_at"
+    t.index ["active_job_id", "created_at"], name: "index_good_jobs_on_active_job_id_and_created_at"
+    t.index ["concurrency_key"], name: "index_good_jobs_on_concurrency_key_when_unfinished", where: "(finished_at IS NULL)"
+    t.index ["cron_key", "created_at"], name: "index_good_jobs_on_cron_key_and_created_at"
+    t.index ["cron_key", "cron_at"], name: "index_good_jobs_on_cron_key_and_cron_at", unique: true
+    t.index ["queue_name", "scheduled_at"], name: "index_good_jobs_on_queue_name_and_scheduled_at", where: "(finished_at IS NULL)"
+    t.index ["scheduled_at"], name: "index_good_jobs_on_scheduled_at", where: "(finished_at IS NULL)"
```
Comment (wojtodzio) [inline db/schema.rb:45]: Is this change intentional?

**Review by wojtodzio — APPROVED** [review]
Nice job!

File: db/schema.rb (hunk 7 above)
Comment (Anakin100100) [reply db/schema.rb:45]: No this was automatic regeneration of the schema during database recreation

**Review by wojtodzio — APPROVED** [review]

**Review by micorix — APPROVED** [review]

**Review by wojtodzio — APPROVED** [review]
I'm not sure why my review was dismissed, but approving again so it can be merged
